cmd/osmosisd/cmd/root.go

```diff
@@ -1034,3 +1058,37 @@ func transformCoinValueToBaseInt(coinValue, coinDenom string, assetMap map[strin
 	}
 	return "", fmt.Errorf("denom %s not found in asset map", coinDenom)
 }
+
+// WriteCustomAppConfigFile first checks the provided config for a special case with the wasm config.
+// This determines what the template should be. Based on this, it writes the new app.toml config file.
+func WriteCustomAppConfigFile(configFilePath string, config CustomAppConfig) {
+	// There is a single wasm config that requires special logic to handle
+	// For some reason, they base a value on whether a line is commented out or not...
+
+	// Check if the simulation_gas_limit line should be commented out (it is either nil or 0)
+	isCommentedOut := config.WasmConfig.SimulationGasLimit == nil || *config.WasmConfig.SimulationGasLimit == 0
+
+	// Modify the template by uncommenting the line if a value is provided for simulation_gas_limit
+	if !isCommentedOut {
+		OsmosisAppTemplate = strings.Replace(OsmosisAppTemplate, "# simulation_gas_limit =", "simulation_gas_limit = {{ .WasmConfig.SimulationGasLimit }}", 1)
+	}
```

Comment:
This is sadly the one "hacky" thing that had to be done. For some reason, wasm utilizes the block gas limit IFF the value is commented out or missing (which is generally what we want the default to be). Therefore, we need to use one template if the user doesn't provide a custom simulation_gas_limit (i.e. a template with the value commented out, because 0 is invalid and we don't want to just delete it from the template), and another template with the line uncommented in the unlikely event a user wants to specify this.